editoast/src/views/timetable/train_schedule.rs

```diff
@@ -817,4 +1195,261 @@ pub mod tests {
+        };
+
+        let train_schedule: Changeset<models::TrainSchedule> = TrainScheduleForm {
+            timetable_id: Some(timetable.id),
+            train_schedule: train_schedule_base.clone(),
+        }
+        .into();
+        let train_schedule_valid = train_schedule
+            .create(&mut db_pool.get_ok())
+            .await
+            .expect("Failed to create train schedule");
+
+        let mut core = MockingClient::new();
+        core.stub("/pathfinding/blocks")
+            .method(reqwest::Method::POST)
+            .response(StatusCode::OK)
+            .json(json!({
+                "status": "not_found_in_tracks"
+            }))
+            .finish();
+        let app = TestAppBuilder::new()
+            .db_pool(db_pool.clone())
+            .core_client(core.into())
+            .build();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+
+        let request = app.post("/train_schedule/track_occupancy").json(
+            &json!({"train_schedule_ids": [train_schedule_valid.id],
+                "operational_point_id": "South_West_station",
+                "infra_id": small_infra.id
+            }),
+        );
+        let response: HashMap<String, Vec<TrackOccupancy>> =
+            app.fetch(request).assert_status(StatusCode::OK).json_into();
+        assert_eq!(response.len(), 0);
+    }
+
+    #[rstest]
+    #[case("T2", 50, 0, 100, Direction::StartToStop, vec![0, 100], vec![0, 1000], true, "T2")]
+    #[case("T2", 150, 0, 100, Direction::StartToStop, vec![0, 100], vec![0, 1000], false, "")]
+    #[case("T2", 50, 0, 100, Direction::StopToStart, vec![0, 100], vec![0, 1000], true, "T2")]
+    fn test_compute_track_occupancy(
+        #[case] track_name: &str,
+        #[case] offset: u64,
+        #[case] begin: u64,
+        #[case] end: u64,
+        #[case] direction: Direction,
+        #[case] positions: Vec<u64>,
+        #[case] times: Vec<u64>,
+        #[case] should_be_some: bool,
+        #[case] expected_track: &str,
+    ) {
+        use editoast_schemas::primitives::Identifier;
+
+        let track_section = Identifier::from(track_name);
+        let track_range = TrackRange {
+            track_section: track_section.clone(),
+            begin,
+            end,
+            direction,
+        };
+        let track_range_vec = vec![track_range];
+        let path_projection = PathProjection::new(&track_range_vec);
+        let operational_point_track_offsets = vec![TrackOffset {
+            track: track_section.clone(),
+            offset,
+        }];
+        let report_train = ReportTrain {
+            positions: positions.clone(),
+            times: times.clone(),
+            speeds: vec![10.0; positions.len()],
+            energy_consumption: 0.0,
+            path_item_times: vec![0, 1000],
+        };
+        let train_schedule = models::TrainSchedule::default();
+
+        let result = compute_track_occupancy(
```

Comment:
shouldn't `compute_track_occupancy_with_path_item_id` be unit tested directly as well? (It is tested through the requests in the tests above though.)